### PR TITLE
NSFS |  Tests refactoring start

### PR DESCRIPTION
**Pull request by romayalon**
### Explain the changes
1. nc_nsfs_manage_entity_types & nc_nsfs_manage_entity_types moved to constants file, replaces all references to the constants file.
2. IS_MAC re use
3. get_tmp_by_os() function and re use
4. moved DEFAULT_FS_CONFIG to a function
5. re-ordered requires.

### Testing Instructions:
1. 

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/util/native_fs_utils.js [hunk 1]
```diff
@@ -433,13 +433,14 @@ function isDirectory(ent) {
...
  * @returns {nb.NativeFSContext}
  */
-function get_process_fs_context(config_root_backend) {
+function get_process_fs_context(config_root_backend, warn_threshold_ms) {
     return {
         uid: process.getuid(),
         gid: process.getgid(),
-        warn_threshold_ms: config.NSFS_WARN_THRESHOLD_MS,
+        warn_threshold_ms: warn_threshold_ms || config.NSFS_WARN_THRESHOLD_MS,
         backend: config_root_backend
```
Comment (shirady) [inline src/util/native_fs_utils.js:444]: The JSDoc indicates (by using the square brackets) that the value `config_root_backend` is optional and we have a default value in the config file, so I suggest using it.
```suggestion
        backend: config_root_backend || config.NSFS_NC_CONFIG_DIR_BACKEND,
```
Then your calls to `get_process_fs_context('', 100)` would change to `get_process_fs_context()` (using the default values).

File: src/util/native_fs_utils.js [hunk 2]
```diff
@@ -433,13 +433,14 @@ function isDirectory(ent) {
 
 /**
  * @param {string} [config_root_backend]
+ * @param {number} [warn_threshold_ms]
  * @returns {nb.NativeFSContext}
```
Comment (shirady) [inline src/util/native_fs_utils.js:437]: Since it is a general function what about the other optional properties `report_fs_stats` `disable_ctime_check`? (of course, the will be undefined, but maybe this function can be changed for those cases (if needed for other parts of the code).

File: src/test/unit_tests/jest_tests/test_nc_nsfs_bucket_cli.test.js [hunk 3]
```diff
@@ -165,13 +158,13 @@ describe('manage nsfs cli bucket flow', () => {
         });
 
         it('cli delete bucket and delete temp dir', async () => {
-            let is_path_exists = await native_fs_utils.is_path_exists(DEFAULT_FS_CONFIG, bucket_temp_dir_path);
-            expect(is_path_exists).toBe(true);
+            let path_exists = await is_path_exists(DEFAULT_FS_CONFIG, bucket_temp_dir_path);
```
Comment (shirady) [inline src/test/unit_tests/jest_tests/test_nc_nsfs_bucket_cli.test.js:161]: I would change the name of the function instead of the name of the variable, for example:
```suggestion
            let is_path_exists = await check_path_exists(DEFAULT_FS_CONFIG, bucket_temp_dir_path);
```

File: src/test/unit_tests/test_bucketspace.js [hunk 4]
```diff
@@ -3,43 +3,37 @@
...
+const os = require('os');
+const _ = require('lodash');
+const path = require('path');
 const util = require('util');
+const http = require('http');
+const mocha = require('mocha');
 const AWS = require('aws-sdk'); // GAP upload is still using AWS SDK V2
+const assert = require('assert');
+const config = require('../../../config');
 const { S3 } = require('@aws-sdk/client-s3');
```
Comment (shirady) [inline src/test/unit_tests/test_bucketspace.js:16]: I guess you also did some order in this part, would you please leave those in sequence (one after another):
```
const AWS = require('aws-sdk'); // GAP upload is still using AWS SDK V2
const { S3 } = require('@aws-sdk/client-s3');
const { NodeHttpHandler } = require("@smithy/node-http-handler");
```

File: src/test/unit_tests/test_bucketspace_fs.js [hunk 5]
```diff
@@ -1,37 +1,34 @@
...
+const NamespaceFS = require('../../sdk/namespace_fs');
+const BucketSpaceFS = require('../../sdk/bucketspace_fs');
 const native_fs_utils = require('../../util/native_fs_utils');
+const { get_tmp_path_by_os } = require('../system_tests/test_utils');
 
-const BucketSpaceFS = require('../../sdk/bucketspace_fs');
-const NamespaceFS = require('../../sdk/namespace_fs');
-const nb_native = require('../../util/nb_native');
 
+const buckets = 'buckets';
```
Comment (shirady) [inline src/test/unit_tests/test_bucketspace_fs.js:20]: Maybe we can also move it to another file (optional are `manage_nsfs_constants.js` or `test_utils`) and use it in the tests?
for example:
```
const DIR_NAME_TYPE = {
    ACCOUNTS: 'accounts',
    BUCKETS: 'buckets',
    ACCESS_KEYS: 'access_keys'
};
```

**Review by shirady — COMMENTED** [review]
Looks good, a couple of comments/questions

**Comment by shirady** [comment]
@romayalon in PR #7776 when I run locally the test: `sudo node ./node_modules/mocha/bin/mocha src/test/unit_tests/test_bucketspace_fs.js` I noticed that there is a test there that its path should be converted so I can run it on Mac, so you will probably need to refactor it some to match the style of this PR changes.

File: src/test/system_tests/test_utils.js [hunk 6]
```diff
@@ -311,6 +296,13 @@ async function delete_fs_user_by_platform(name) {
     }
 }
 
+/**
+ * get_tmp_path_by_os returns the tmp path by the process platform
+ */
```
[the remark this answers is not in the document]
Comment (romayalon) [reply src/test/system_tests/test_utils.js:301]: I decided to change this function to a constant and not accept any arg, see 
`const TMP_PATH = os_utils.IS_MAC ? '/private/tmp/' : '/tmp/';`
comment is not relevant anymore, resolving

File: src/util/native_fs_utils.js (hunk 1 above)
Comment (romayalon) [reply src/util/native_fs_utils.js:444]: get_process_fs_context() is being called from various files, not in all of them config.NSFS_NC_CONFIG_DIR_BACKEND is the default, I changed the parameter name from config_root_backend to backend so it'll be more understood

File: src/test/unit_tests/jest_tests/test_nc_nsfs_bucket_cli.test.js (hunk 3 above)
Comment (romayalon) [reply src/test/unit_tests/jest_tests/test_nc_nsfs_bucket_cli.test.js:161]: Liran asked Naveen to change it to this name a few PRs ago, and I also prefer is_... as predicates names

File: src/test/unit_tests/test_bucketspace.js (hunk 4 above)
Comment (romayalon) [reply src/test/unit_tests/test_bucketspace.js:16]: np, done

File: src/test/unit_tests/test_bucketspace_fs.js (hunk 5 above)
Comment (romayalon) [reply src/test/unit_tests/test_bucketspace_fs.js:20]: Like!

**Comment by romayalon** [comment]
@shirady can you please share the exact test name?

File: src/util/native_fs_utils.js (hunk 2 above)
Comment (romayalon) [reply src/util/native_fs_utils.js:438]: disable_ctime_check is not relevant, I added report_fs_stats, thanks

File: src/util/native_fs_utils.js [hunk 7]
```diff
@@ -432,15 +432,18 @@ function isDirectory(ent) {
...
-function get_process_fs_context(config_root_backend) {
+function get_process_fs_context(backend, warn_threshold_ms, report_fs_stats) {
     return {
         uid: process.getuid(),
         gid: process.getgid(),
-        warn_threshold_ms: config.NSFS_WARN_THRESHOLD_MS,
-        backend: config_root_backend
+        warn_threshold_ms: warn_threshold_ms || config.NSFS_WARN_THRESHOLD_MS,
+        backend: backend,
+        report_fs_stats
```
Comment (shirady) [inline src/util/native_fs_utils.js:446]: Does the name of the function is always `report_fs_stats`?
Did understand why it is not:
```suggestion
        report_fs_stats: report_fs_stats
```

File: src/test/system_tests/test_utils.js [hunk 8]
```diff
@@ -311,6 +303,8 @@ async function delete_fs_user_by_platform(name) {
     }
 }
 
+
+
```
Comment (shirady) [inline src/test/system_tests/test_utils.js:307]: Did you add those newlines on purpose?

File: src/test/unit_tests/test_namespace_fs_mpu.js [hunk 9]
```diff
@@ -3,33 +3,31 @@
...
-const MAC_PLATFORM = 'darwin';
 const DUMMY_OBJECT_SDK = make_DUMMY_OBJECT_SDK();
 
 const src_bkt = 'src';
-let tmp_fs_path = '/tmp/test_namespace_fs_mpu';
-if (process.platform === MAC_PLATFORM) {
-    tmp_fs_path = '/private/' + tmp_fs_path;
-}
+const tmp_fs_path = path.join(TMP_PATH, 'test_namespace_fs_mpu');
+
```
Comment (shirady) [inline src/test/unit_tests/test_namespace_fs_mpu.js:30]: Did you add this newline on purpose?

**Review by shirady — COMMENTED** [review]
Last minor questions

File: src/util/native_fs_utils.js (hunk 1 above)
Comment (shirady) [reply src/util/native_fs_utils.js]: What about changing your calls from `get_process_fs_context('', 100)` to `get_process_fs_context()` (using the default values)?

**Comment by shirady** [comment]
> @shirady can you please share the exact test name?

@romayalon Sure, it is `delete_bucket for should_create_underlying_storage false`. here is the place where we would need to refactor:
https://github.com/noobaa/noobaa-core/blob/3142e2b7dc98923858702b309b8df2b203a413e2/src/test/unit_tests/test_bucketspace_fs.js#L372-L373

File: src/test/unit_tests/jest_tests/test_nc_nsfs_bucket_cli.test.js (hunk 3 above)
Comment (shirady) [reply src/test/unit_tests/jest_tests/test_nc_nsfs_bucket_cli.test.js:156]: - I saw that the function was changed from `config_file_exists()` to `is_path_exists()` in PR #7718 since the first name was not accurate to the function functionality.
- Adding a verb ('check_'/'validate_', etc.) instead of the 'is_' in the name of the function is just a matter of style.

File: src/util/native_fs_utils.js (hunk 7 above)
Comment (romayalon) [reply src/util/native_fs_utils.js:446]: not sure I understood the question... what's the difference between the current line and the suggested one?
```
> const x = 1;
> console.log({x : x});
{ x: 1 }
undefined
> console.log({x});
{ x: 1 }
```

Comment (shirady) [reply src/util/native_fs_utils.js:446]: I don't remember what I meant, sorry, resolving.

File: src/util/native_fs_utils.js (hunk 1 above)
Comment (romayalon) [reply src/util/native_fs_utils.js]: done, thanks!

File: src/test/unit_tests/test_bucketspace.js [hunk 10]
```diff
@@ -478,9 +472,10 @@ mocha.describe('bucket operations - namespace_fs', async function() {
         assert.ok(list_ok);
     });
     mocha.it('list buckets with dn', async function() {
+        // Give s3_correct_uid access to the required buckets
         const res = await s3_correct_dn_default_nsr.listBuckets({});
         console.log(inspect(res));
-        const list_ok = bucket_in_list([bucket_name + '-s3-2'], [], res.Buckets);
+        const list_ok = bucket_in_list([bucket_name + "-s3-2"], [], res.Buckets);
```
Comment (shirady) [inline src/test/unit_tests/test_bucketspace.js:478]: Why did you change it to double qoutes?

File: src/test/unit_tests/test_bucketspace_versioning.js [hunk 11]
```diff
@@ -8,41 +8,34 @@ const path = require('path');
 const http = require('http');
 const mocha = require('mocha');
 const { S3 } = require('@aws-sdk/client-s3');
```
Comment (shirady) [inline src/test/unit_tests/test_bucketspace_versioning.js:10]: If you can leave the `const { S3 } = require('@aws-sdk/client-s3');` and `const { NodeHttpHandler } = require('@smithy/node-http-handler');` near.

File: src/test/unit_tests/test_nc_nsfs_cli.js [hunk 12]
```diff
@@ -309,19 +293,19 @@ mocha.describe('manage_nsfs cli', function() {
             }
         });
 
-        mocha.it('cli bucket update bucket owner', async function() {
-            const action = nc_nsfs_manage_actions.UPDATE;
+        mocha.it('cli bucket update owner email', async function() {
```
Comment (shirady) [inline src/test/unit_tests/test_nc_nsfs_cli.js:296]: Why did you change it back to email?

**Review by shirady — APPROVED** [review]
LGTM
only minor questions

File: src/test/unit_tests/test_bucketspace.js (hunk 10 above)
Comment (romayalon) [reply src/test/unit_tests/test_bucketspace.js]: it came back during conflicts resolutions

File: src/test/unit_tests/test_nc_nsfs_cli.js (hunk 12 above)
Comment (romayalon) [reply src/test/unit_tests/test_nc_nsfs_cli.js]: it came back during conflicts resolutions
